vuepress/docs/docs/interop.md

```diff
@@ -126,23 +133,30 @@ val addBook: Endpoint[(Book, String), Nothing, Unit, Nothing] =
 ```
 
 And a possible implementation:
-```scala
+```scala mdoc:silent
+import zio.UIO
+
 def bookAddLogic(book: Book, token: String): UIO[Unit] = ???
 ```
 
 Just like you can create an http4s route by calling `toRoute` and passing an implementation, call `toGraphQL` to create a GraphQL API:
-```scala
+```scala mdoc:silent
+import caliban.GraphQL
+import caliban.interop.tapir._ // summons 'toGraphQL' extension
+
 val api: GraphQL[Any] = addBook.toGraphQL((bookAddLogic _).tupled)
 ```
 That's it! You can combine multiple `GraphQL` objects using `|+|` and expose the result using one of Caliban's adapters.
 
-If you want to reuse `bookAddLogic` for both GraphQL and regular HTTP, you can turn your `Endpoint` into a `ServerEndpoint` by calling `.serverLogic`:
-```scala
-val addBookEndpoint: ServerEndpoint[(Book, String), Nothing, Unit, Nothing, UIO] =
-  addBook.serverLogic[UIO] { case (book, token) => bookAddLogic(book, token).either }
+If you want to reuse `bookAddLogic` for both GraphQL and regular HTTP, you can turn your `Endpoint` into a `ServerEndpoint` by calling `.serverLogicSuccess`:
+```scala mdoc:silent
+import sttp.tapir.server.ServerEndpoint
+
+val addBookEndpoint: ServerEndpoint.Full[Unit, Unit, (Book, String), Nothing, Unit, Any, UIO] =
+  addBook.serverLogicSuccess[UIO] { case (book, token) => bookAddLogic(book, token) }
```

Comment:
If we want to keep using `addBook.serverLogic` in the example, the effect type should be changed (since `UIO` cannot fail).

[tapir/ExampleApp](https://github.com/ghostdogpr/caliban/blob/master/examples/src/main/scala/example/tapir/ExampleApp.scala) uses `IO[String, Unit]` for demonstration purposes.